my-app/src/api/api.tsx

```diff
@@ -1,69 +1,64 @@
-import axios from 'axios';
-import type { LatLngTuple } from '../types';
-
-import type { RouteResponse } from '../types';
+import axios from 'axios'
+import type { LatLngTuple } from '../types'
+import type { RouteResponse } from '../types'
 
 interface ApiRouteRequestPayload {
-    coordinates: {
-        lat: number;
-        lon: number;
-    }[];
+    coordinates: { lat: number; lon: number }[]
+    avoid_zones?: { lon: number; lat: number }[][]
 }
 
-class ApiService {
-    private baseURL: string;
-
+export class ApiService {
+    private baseURL: string
     constructor(baseURL: string = 'http://localhost:8000') {
-        this.baseURL = baseURL;
+        this.baseURL = baseURL
     }
 
     /**
-     * Fetch route between multiple points (start, waypoints, end).
-     * @param points An array of [lat, lng] tuples. Must contain at least a start and end point.
-     * @returns Promise resolving to array of [lng, lat] coordinates for the full route.
+     * Fetch route between multiple points, avoiding any restricted polygons.
+     *
+     * @param points An array of [lat, lng] tuples. Must contain at least start & end.
+     * @param avoidZones Optional array of polygons, each polygon is an array of [lat, lng].
      */
-    async fetchRoute(points: LatLngTuple[]): Promise<LatLngTuple[]> {
+    async fetchRoute(
+        points: LatLngTuple[],
+        avoidZones: LatLngTuple[][] = []
+    ): Promise<LatLngTuple[]> {
         if (points.length < 2) {
-            throw new Error("At least two points (start and end) are required to fetch a route.");
+            throw new Error('At least two points (start and end) are required.')
+        }
+        // map to API‐friendly keys
+        const apiCoords = points.map(([lat, lon]) => ({ lat, lon }))
```

Comment:
No need to do this conversion here. Your server should know how to handle these key differences